pkg/cmd/pulumi/new.go

```diff
@@ -142,6 +165,44 @@ func runNew(ctx context.Context, args newArgs) error {
 		}
 	}
 
+	if args.templateNameOrURL == "" {
+		// Try to read the current project
+		project, _, err := readProject()
+		if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
+			return err
+		}
+
+		b, err := currentBackend(ctx, project, opts)
+		if err != nil {
+			return err
+		}
+
+		name, _, _, err := b.CurrentUser()
+		if err != nil {
+			return err
+		}
+		var aiOrTemplate string
+		if shouldPromptForAIOrTemplate(args, b) {
+			aiOrTemplate, err = chooseWithAIOrTemplate(opts)
+		} else {
+			aiOrTemplate = deriveAIOrTemplate(args)
+		}
+		if err != nil {
+			return err
+		}
+		if aiOrTemplate == "ai" {
+			if b.Name() != "pulumi.com" {
```

Comment:
Do we want AI prompts to work for other instances of Pulumi Cloud, e.g. staging, dev stacks, etc? Or only the production service?

Reply:
That's a great question - I _think_, because this feature doesn't actually interact with p-cloud (all the integrations are with the AI service directly) that leaving it as the production service makes sense. I'd be more open to configuring this / being a bit more clever here if we wanted to test upstream integrations with other backends, but that's not really relevant here.